### ISSUE TITLE
Tensors on 2 devices CPU and cuda

### ISSUE THREAD

**Issue by nikhilchh** [Daniil-Osokin/lightweight-human-pose-estimation.pytorch#208]
When I try to retrain the model WITHOUT --weight-only flag I get the following issue:
"Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu!"

My guess it is something to do with optimizer or scheduler. As they are loaded from saved checkpoint when I don't turn on weight-only flag.

**Comment by Daniil-Osokin** [comment]
Hi! So what is your issue?

**Comment by nikhilchh** [comment]
The issue is that you can retrain using the existing checkpoint when u use -weights-only flag.
If you don't use -weights-only flag it crashes at optimizer step.

And not using -weights-only flag is a legit use case.

**Comment by Daniil-Osokin** [comment]
Did you follow step 3-5 from the README.md and observe this issue?

**Comment by nikhilchh** [comment]
Not. Because there you have used -weights-only flag. 
But as I said not using -weights-only flag is a valid use case. As one would want to start from same:
-Learning rates
-Epoch number
which are saved in the last checkpoint.

**Comment by Daniil-Osokin** [comment]
Do you need a help with this?

**Comment by nikhilchh** [comment]
Yeah. It would be nice.
I want to pick up the training from where I stopped and also want to know the epoch and iteration number from saved checkpoint. But I can't do that as when i dont use --weight-only flag I get an error.
Can you try to replicate the problem on your side ?
Just load the model 370000.pth without using --weight-only flag.

**Comment by Daniil-Osokin** [comment]
Try to load network on a gpu before passing parameters to optimizer, just append [here](https://github.com/Daniil-Osokin/lightweight-human-pose-estimation.pytorch/blob/master/train.py#L26) `.cuda()` call and try to continue training from checkpoint without `--weights-only` flag.

**Comment by Daniil-Osokin** [comment]
Hope, it helped.

**Comment by nikhilchh** [comment]
Thanks. It worked.
